general-concepts/slotting/text.xml

```diff
@@ -60,8 +60,104 @@ automatically rebuilt</uri> when the subslot of a runtime dependency changes.
+</p>
+
+<p>
+You may wish to review the QA team's documentation on subslots on the
+<uri link="https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Subslots">
+wiki.</uri>
+</p>
+
+<note>
+Care must be taken when using sub-slots in a library ebuild for the first time.
+Adding sub-slots will trigger rebuilds for all the packages that already use sub-slot
+dependencies (e.g. Switching from SLOT="0" to SLOT="0/14" in <c>media-libs/libpng</c> and
+package <c>foo</c> depends on <c>libpng:0=</c>).
+Therefore, it's best if you start using sub-slots in the library when the existing library
+interface changes.
+</note>
+</body>
+
+<subsection>
+<title>ABI breakage</title>
+<body>
+
+<p>
+There are two ways a library can break compatibility with its consumers.
+</p>
+
+<ul>
+<li>
+ABI (Application Binary Interface): this affects binaries built before the change.
+Applications linked against a library pre-ABI break may not work correctly after
+the break. These changes are related to internal structure, such as the size of
+a <c>struct</c> or the type of an argument (e.g. integer width). Fixing this
+requires a <e>rebuild</e> of all consumers.
+</li>
+<li>
+API (Application Program Interface): this affects consumers at compile time and
+usually occurs when a library has deprecated and then removed a function. Fixing
+this requires a <e>code</e> change in consumers.
+</li>
+</ul>
+
+<p>
+Note that subslots are not exclusively for this purpose. While they form
+the majority of uses in the Gentoo tree, subslots may have a meaning that
+is completely divorced from SONAMEs or ABI breakage enetirely. Check
+the usage in the relevant packages before using a subslot operator!
+</p>
+
+<p>
+When we become aware of ABI breakage, we will increment the subslot. Note that
+the subslot does not have to strictly be the SONAME and therefore could be
+an arbitrary integer (or even string).
+</p>
+
+<p>
+Be aware that some upstreams may make releases without verifying if binary
+compatibility has been broken in a minor release. You should check using
+tools like <c>libabigail</c> or <e>ABI Laboratory</e>.
+</p>
+
+<p>
+Generally, consumers <b>which link against</b> a library possessing a subslot
+that <e>represents SONAME or binary compatibility</e> should subscribe to
+it (request to be rebuilt when the subslot changes) with <c>:=</c>. Also see
+the QA Policy Guide for information on
+<uri link="https://projects.gentoo.org/qa/policy-guide/dependencies.html#proactive-use-of-slot-operators">
+proactively subscribing to subslots before they are defined.
+</uri>
+</p>
+</body>
+
+<subsubsection>
+<title>General naming of a sub-slot</title>
+<body>
+
+<p>
+The convention is that libraries express binary compatibility via <c>SONAME</c>,
+e.g. <c>libfoo.so.1.2</c>. The expectation would be (unless stated otherwise
+by upstream) that all releases with the same major SONAME (1 here) have
+binary compatibility, while the minor SONAME (2) may be incremented on
```

Comment:
technically incorrect (because too strong), SONAME guarantees updates only in one direction